presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/glue/TestHiveGlueMetastore.java

```diff
@@ -41,7 +55,8 @@ protected HiveMetastore createMetastore(File tempDir)
         GlueHiveMetastoreConfig glueConfig = new GlueHiveMetastoreConfig();
         glueConfig.setDefaultWarehouseDir(tempDir.toURI().toString());
 
-        return new GlueHiveMetastore(HDFS_ENVIRONMENT, glueConfig);
+        Executor executor = new BoundedExecutor(newCachedThreadPool(daemonThreadsNamed("hive-glue-%s")), 10);
```

Comment:
Create a shared cached thread pool for the test class using `@BeforeClass` / `@AfterClass` so that we shut it down, otherwise we can run out of JVM threads when running many tests. See `TestThrottledAsyncQueue` for an example.

Reply:
Will do.